app/src/main/java/io/appium/uiautomator2/server/ServerConfig.java

```diff
@@ -16,10 +16,106 @@
 
 package io.appium.uiautomator2.server;
 
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
 public class ServerConfig {
-    private final static int PORT = 6790;
+    // server port (default: `6790`)
+    private final static int SERVER_PORT =
+        System.getenv("SERVER_PORT") == null ? 6790 :
+            Integer.parseInt(System.getenv("SERVER_PORT"));
+    // mjpeg server port (default: `7810`)
+    private final static int MJPEG_SERVER_PORT =
+        System.getenv("MJPEG_SERVER_PORT") == null ? 7810 :
+            Integer.parseInt(System.getenv("MJPEG_SERVER_PORT"));
+    // mjpeg stream will try to match this framerate (default: `10`)
+    private final static int MJPEG_SERVER_FRAMERATE =
+        System.getenv("MJPEG_SERVER_FRAMERATE") == null ? 10 :
+            Integer.parseInt(System.getenv("MJPEG_SERVER_FRAMERATE"));
+    // mjpeg scale should be between 1 and 100 (default: `50`)
+    private final static int MJPEG_SCALING_FACTOR =
+        System.getenv("MJPEG_SCALING_FACTOR") == null ? 50 :
+            Integer.parseInt(System.getenv("MJPEG_SCALING_FACTOR"));
+    // mjpeg quality is between 0 and 100 (default: `75`)
+    private final static int MJPEG_SERVER_SCREENSHOT_QUALITY =
+        System.getenv("MJPEG_SERVER_SCREENSHOT_QUALITY") == null ? 75 :
+            Integer.parseInt(System.getenv("MJPEG_SERVER_SCREENSHOT_QUALITY"));
+    // if filtering should be used in mjpeg resize operation (default: `false`)
+    private final static boolean MJPEG_FILTERING =
+        System.getenv("MJPEG_FILTERING") == null ? false :
+            System.getenv("MJPEG_FILTERING") == "true";
+
+    // In-memory overrides
+    private static ConcurrentMap<String, Object> overrides =
+        new ConcurrentHashMap<>();
+
+    /* Getter(s) */
 
     public static int getServerPort() {
-        return PORT;
+        if (overrides.containsKey("serverPort")) {
+            return (int) overrides.get("serverPort");
```

Comment:
please avoid using magic strings. Use setting name property if possible

Reply:
This will be resolved w/ the upcoming update(s)